File: top-interview-questions-easy/list/LeetCode206.cc
#include<iostream>
using namespace std;

struct ListNode{
    int val;
    ListNode *next;
    ListNode() : val(0),next(nullptr) {}
    ListNode(int x): val(x),next(nullptr){}
    ListNode(int x, ListNode *next): val(x),next(next){}
};


class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* cur = NULL, *pre = head;
        while(pre != NULL){
            ListNode *temp = pre->next;
            pre->next = cur;
            cur = pre;
            pre = temp;
        }
        return cur;
    }
};

File: Offer/golang/linkedList/LeetCode206.go
package linkedList

func reverseListLeetCode(head *ListNode) *ListNode {
	if head == nil {
		return head
	}
	cur := head
	var pre *ListNode
	for cur != nil {
		temp := cur.Next
		cur.Next = pre
		pre = cur
		cur = temp
	}
	return pre
}
